src/main/scala/me/shadaj/romeojuliet/Tragedy.scala

```diff
@@ -94,15 +146,7 @@ class Juliet(friar: ActorRef) extends Actor {
           |Romeo, Romeo, Romeo! Here’s drink. I drink to thee.
           |""".stripMargin)
       context.become(sleeping)
-      context.system.scheduler.scheduleOnce(10 second) {
-        context.become(loving(lover))
```

Comment:
Ouch! Thou must not close over actor state when thou execute some code in another thread. `context` is considered actor state. This produced a `NullPointerException` during shutdown – not too dramatic in this case, but in real life there will be dragons.